### ISSUE TITLE
synchronisation data get from multiple tables

### ISSUE THREAD

**Issue by yatinyugtia** [brody2consult/cordova-sqlite-storage#677]
this.db.transaction(function(tx) {
        console.log('tx object');
        console.log(tx);
        console.log('end tx object');
        tx.executeSql("SELECT db_id FROM claims where sync = ?", ['0'],
          function(tx, res) {
            console.log(res);
            var syncClaimList = {};
            var i = 0;
            var call_counter = 0;
            var temp = [];
            while (i < res['rows']['length']) {

              var intern = $.Deferred();
              var intern_test = [];
              intern_test.push($.Deferred());
              console.log('intern_test');
              console.log(intern_test);
              console.log('end intern_test');
              var claims = res['rows']['item'](i);
              var claim_id = claims['db_id'];
              var claimnotes = [];
              var j = 0;

              temp.push(angular.copy(claim_id));
              intern_test[i].then(
                tx.executeSql("SELECT * FROM claim_internalnote WHERE claim_id = ? and sync='0'", [claim_id],
                  function(tx, internalnote) {
                    var response = [];
                    console.log('claim_id :' + claim_id);

                    while (j < internalnote['rows']['length']) {
                      claimnotes.push(internalnote['rows']['item'](j));
                      j = j + 1;
                    }
                    return intern_test[call_counter].resolve(claimnotes);
                  })
              );

              var intern_promise = [];
              intern_promise.puch(intern[i].promise());
              //for (temp1 = 1; temp1 <= temp.length; temp1++) {
              intern_promise[i].then(function(result) {

                console.log(temp[call_counter] + ' claim id');

                syncClaimList = {
                  'claim_id': temp[call_counter],
                  'claimnotes': result
                };
                call_counter = call_counter + 1;
                console.log(syncClaimList);
                var pdata = jQuery.param({
                  'user_id': localStorage.getItem("UserId"),
                  'user_token': localStorage.getItem("UserToken"),
                  'user_type': localStorage.getItem("UserType"),
                  'synchronization_data': JSON.stringify(syncClaimList),
                  'device_token': ''
                });
                var urlBase = Api.urlBase + "sync";
                req = {
                  headers: Api.HEADERS
                };
                $http.post(urlBase, pdata, req).then(
                  function(response) {
                    console.log('syncing issue');
                    console.log(response);
                    console.log('end syncing issue');
                  },
                  function(error) {
                    console.log('syncing issue');
                  }
                )
                //return d.resolve(syncClaimList);
              });
              i = i + 1;
            }
          },
          function(tx, error) {
            console.log(error);
          });
      });

    }

**Comment by brody2consult** [comment]
My apologies for the delay. The code looks really hard to follow, especially with the lack of proper JavaScript highlighting. There is also lack of information about how the data may be populated.

Please be sure to keep the reproduction code brief, clear, and complete in the future.

Private support is also available at: <mailto:sales@litehelpers.net>
